### PR TITLE
improving error handling for createDatasetForUser

### PR DESCRIPTION
**Pull request by gniezen**
The existing error handling returns `null` for POST errors, so I've changed it to return the correct HTTP error, or create a new error if the HTTP response is unexpected.

Ping @darinkrauss for review.

### CODE REVIEW

**Comment by gniezen** [comment]
@darinkrauss I've just updated this again to work with latest changes to master.

File: index.js [hunk 1]
```diff
@@ -373,11 +373,19 @@ module.exports = function (config, deps) {
         .set(common.SESSION_TOKEN_HEADER, user.getUserToken())
         .end(
         function (err, res) {
+          console.log("err", err, "res", res);
```
Comment (darinkrauss) [inline index.js]: If we are going to have a console log message do we want something more descriptive, like console.log("Response from createDatasetForUser: err=", err, ", res=", res);`?

File: index.js [hunk 2]
```diff
@@ -373,11 +373,19 @@ module.exports = function (config, deps) {
...
+          } else if (res.error === true) {
+            if(_.isObject(res.body)) {
+              return cb(res.body); // for our custom error arrays
+            } else {
+              return cb(res.error);
+            }
           } else if (res.status !== 201) {
-            return cb(res.body);
+            var errText = 'Unexpected HTTP response: ' + res.status;
+            return cb(new Error(errText));
```
Comment (darinkrauss) [inline index.js]: I'd just do: `return cb(new Error('Unexpected HTTP response: ' + res.status));`

**Comment by darinkrauss** [comment]
A couple really minor nits, but LGTM.  I do notice that the Travis CI build failed, though. Might want to look into that.

File: index.js (hunk 1 above)
Comment (gniezen) [inline index.js]: Arrgh, I had that in there for debugging - didn't mean to push it up, will remove.

**Comment by gniezen** [comment]
@darinkrauss Travis CI build has been failing on platform-client for quite some time now. Maybe @jh-bate knows why? Easy fix, or something for the next sprint?

**Comment by gniezen** [comment]
Merging, as failing tests are unrelated to the PR

**Pull request merged**